bundles/org.eclipse.e4.ui.workbench.renderers.swt/src/org/eclipse/e4/ui/workbench/renderers/swt/CTabRendering.java

```diff
@@ -1330,7 +1335,9 @@ private IEclipsePreferences getSwtRendererPreferences() {
 
 	private boolean isPartOfEditorStack() {
 		MUIElement element = (MUIElement) parent.getData(AbstractPartRenderer.OWNING_ME);
-		return element != null && element.getTags().contains(EditorTag);
+		int location = eModelService.getElementLocation(element);
+		return (element != null && element.getTags().contains(EditorTag))
+				|| ((location & EModelService.IN_SHARED_AREA) != 0);
```

Comment:
I don't think this is correct, there can be many `Area`(s) in a RCP application and kust because they are shared don't make them the editor area.

Reply:
> It is obviously not the same check ;-)

Really?

`(location & EModelService.IN_SHARED_AREA) != 0)` and
`(location & EModelService.IN_SHARED_AREA) != 0)` look quite identical for me. Am I overlooking something?

Reply:
`(element != null && element.getTags().contains(EditorTag))`
is not the same as 
`(location & EModelService.IN_SHARED_AREA) != 0)`

so who ever did the tag check instead of the location check has did this probably for a purpose, now the EditorTag check even seem to become obsolete as the EditorArea is almost always shared (I can't think about a reason why one don't want this), also as I mentioned Eclipse uses the ID to determine it is "the editor stack" so why not consider this as well and so on.

Reply:
The following works for me:
```
	private boolean isPartOfEditorStack() {
		MUIElement element = (MUIElement) parent.getData(AbstractPartRenderer.OWNING_ME);
		EObject root = EcoreUtil.getRootContainer((EObject) element, true);
		if (root instanceof MContext context) {
			EModelService eModelService = context.getContext().get(EModelService.class);
			if (eModelService != null) {
				int location = eModelService.getElementLocation(element);
				return (location & EModelService.IN_SHARED_AREA) != 0;
			}
		}
		return false;
	}
```
EcoreUtil.getRootContainer is null safe.

It looks like this where otherwise all the tabs in both areas would be squished:

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/208716/a5f92d80-b62e-4865-936f-617ba0a26b3d)

Reply:
> The following works for me

So does this mean we have consensus that the PR with the changes proposed by @merks are good and would fix the issue (even when it does not fix the root cause of the disappearing tag).

I have the impression that the issue with the disappearing tag already exists longer (did somebody test with the previous release?) and that the changes to the tab rendering "only" make it visible. Do you agree?

Reply:
Given that this can be demonstrated to fix the current problem, I _think_ the worst thing that could happen is that in some places the tabs might not be squished where you guys want them squished.  But my opinion is just one opinion...